src/contour/TerminalSessionManager.h

```diff
@@ -50,32 +52,38 @@ class TerminalSessionManager: public QAbstractListModel
     [[nodiscard]] int count() const noexcept { return static_cast<int>(_sessions.size()); }
 
     void updateColorPreference(vtbackend::ColorPreference const& preference);
-    display::TerminalDisplay* display = nullptr;
-    TerminalSession* getSession() { return _sessions[0]; }
+
+    display::TerminalDisplay* activeDisplay = nullptr;
+    void FocusOnDisplay(display::TerminalDisplay* display);
 
     void update() { updateStatusLine(); }
 
+    void allowCreation() { _allowCreation = true; }
+
   private:
     contour::TerminalSession* activateSession(TerminalSession* session, bool isNewSession = false);
     std::unique_ptr<vtpty::Pty> createPty(std::optional<std::string> cwd);
 
+    void tryFindSessionForDisplayOrClose();
+
     [[nodiscard]] std::optional<std::size_t> getSessionIndexOf(TerminalSession* session) const noexcept
     {
         if (auto const i = std::ranges::find(_sessions, session); i != _sessions.end())
             return static_cast<std::size_t>(std::distance(_sessions.begin(), i));
         return std::nullopt;
     }
 
-    [[nodiscard]] auto getCurrentSessionIndex() const noexcept
+    [[nodiscard]] auto getCurrentSessionIndex() noexcept
     {
-        return getSessionIndexOf(_activeSession).value();
+        // TODO cache this value
+        return getSessionIndexOf(_displayStates[activeDisplay]).value();
     }
 
     void updateStatusLine()
     {
-        if (!_activeSession)
+        if (!_displayStates[activeDisplay])
             return;
-        _activeSession->terminal().setGuiTabInfoForStatusLine(vtbackend::TabsInfo {
+        _displayStates[activeDisplay]->terminal().setGuiTabInfoForStatusLine(vtbackend::TabsInfo {
```

Comment:
```suggestion
        if (auto* displayState = _displayStates[activeDisplay]; displayState->terminal().setGuiTabInfoForStatusLine(vtbackend::TabsInfo {
```

Avoid the double-lookup to `_displayStates[activeDisplay]`.